the-collab-lab/tcl-37-smart-shopping-list
# Duplicate Item Alert

## Description



In this PR, we updated the `handleSubmit()` function in `AddItemView.jsx` to notify the user if the item they inputted in the form already exists, while considering its letter case or added special characters.

To accomplish this, we used the collection from the database to create an array of only the item values. We then mapped through the item values to make them lowercase and remove any special characters. We applied this change to the input value as well, then checked if the input value exists in the array we created. 

If it exists, we return a toast notification saying "Item already exists!" If it doesn't exist, the item is added to the database collection. 


## Related Issue



Closes #6 

## Acceptance Criteria



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before


![preview-before](https://user-images.githubusercontent.com/82783679/151069524-3c7a2759-951f-4c63-a482-a959d04a39be.gif)


### After


![preview-after](https://user-images.githubusercontent.com/82783679/151069621-218802cb-1952-414d-aac5-949865bfa162.gif)


## Testing Steps / QA Criteria
1. Add an item to the form: "mangoes"
2. Add the same item with uppercase letters & special characters: "MaNGoEs!@#$"
3. You should expect the notification "Item already exists!"
